pkg/provider/azure_loadbalancer.go

```diff
@@ -821,20 +821,58 @@ func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service,
 	return "", false, fmt.Errorf("user supplied IP Address %s was not found in resource group %s", loadBalancerIP, pipResourceGroup)
 }
 
-func (az *Cloud) findMatchedPIPByLoadBalancerIP(service *v1.Service, loadBalancerIP, pipResourceGroup string) (*network.PublicIPAddress, error) {
+func (az *Cloud) findMatchedPIP(loadBalancerIP, pipName, pipResourceGroup string) (pip *network.PublicIPAddress, err error) {
```

Comment:
I checked the usage and it seems `findMatchedPIP` finds PIP with LB IP or PIP name. How about returning error if both 2 variables are set or both unset?

Reply:
We can cover it in the doc that ip will overweight the name.

Reply:
Yes, doc is good for both set.
How about both empty? I think at least a warning is needed. When somehow 2 empty values are passed in, it returns an empty PIP. It will add difficulty when debugging.

Reply:
They can't be both empty, we guard outside of the func.